core/src/Cuda/Kokkos_Cuda_Parallel.hpp

```diff
@@ -663,6 +664,42 @@ class ParallelFor<FunctorType, Kokkos::MDRangePolicy<Traits...>, Kokkos::Cuda> {
       : m_functor(arg_functor), m_rp(arg_policy) {}
 };
 
+__device__ inline int64_t cuda_get_scratch_index(Cuda::size_type league_size,
+                                                 int32_t* scratch_locks) {
+  int64_t threadid = 0;
+  __shared__ int64_t base_thread_id;
+  if (threadIdx.x == 0 && threadIdx.y == 0) {
+    int64_t const wraparound_len = Kokkos::Experimental::min(
+        int64_t(league_size),
+        (int64_t(Kokkos::Impl::g_device_cuda_lock_arrays.n)) /
+            (blockDim.x * blockDim.y));
+    threadid = (blockIdx.x * blockDim.z + threadIdx.z) % wraparound_len;
+    threadid *= blockDim.x * blockDim.y;
+    int done = 0;
```

Comment:
Is this function only ever executed by a single thread? If not, shouldn't `done` be volatile?

Reply:
This is a local variable so I don't think volatile is relevant.